src/Nerdbank.MessagePack.Analyzers/KeyAttributeUseAnalyzer.cs

```diff
@@ -152,6 +210,13 @@ private bool IsMemberSerialized(ISymbol member, ImmutableHashSet<string> ctorPar
 			IFieldSymbol f => f.IsReadOnly,
 			_ => false,
 		};
+
+		// Read-only properties can still be serialized if they are collection types that support IDeserializeInto<T>
+		if (isReadOnly && member is IPropertySymbol property)
```

Comment:
We should allow members to be `IFieldSymbol` as well.
And please add a test or beef up some of your existing new tests for fields as well